File: moforum-web/src/api/board.ts
import { http } from './http'

export type BoardRow = {
  id: number
  name: string
  description?: string
  avatar?: string
  sortOrder?: number
  postCount?: number
  createTime?: string
  updateTime?: string
}

export function listBoards() {
  return http.get<BoardRow[]>('/board/list')
}

export function getBoard(id: number) {
  return http.get<BoardRow>('/board/detail', { params: { id } })
}

File: moforum/src/main/java/com/moforum/entity/Board.java
package com.moforum.entity;

import lombok.Data;

import java.time.LocalDateTime;

@Data
public class Board {
    private Long id;
    private String name;
    private String description;
    private String avatar;
    private Integer sortOrder;
    private Integer postCount;
    private LocalDateTime createTime;
    private LocalDateTime updateTime;
}
